File: e2e/csharp/tests/CapsulePassthroughTests.cs
// Hand-written capsule-passthrough test (not fixture-generated); preserved across regen.

using System;
using Xunit;
using TreeSitterLanguagePack;

namespace TreeSitterLanguagePack
{
    /// <summary>
    /// Verify that TreeSitterLanguagePackConverter.GetLanguage returns a host-native
    /// TreeSitter.Language usable with the upstream TreeSitter.DotNet parser.
    ///
    /// This tests the capsule passthrough feature (#143): the host-native Language
    /// wrapper allows direct interop with third-party tree-sitter consumers.
    /// </summary>
    public class CapsulePassthroughTests
    {
        [Fact]
        public void Test_ParsePythonWithHostLanguage()
        {
            // Get host-native Language via TreeSitterLanguagePackConverter
            var language = TreeSitterLanguagePackConverter.GetLanguage("python");
            Assert.NotNull(language);

            // Create upstream TreeSitter.DotNet parser
            using var parser = new TreeSitter.Parser();

            // Set the host-native language on the parser
            parser.Language = language;

            // Parse Python source
            var source = "def greet(name):\n    return name\n";
            using var tree = parser.Parse(source);

            var rootNode = tree.RootNode;
            Assert.NotNull(rootNode);

            // Verify the root node type is "module" (Python AST root)
            Assert.Equal("module", rootNode.Type);
        }

        [Fact]
        public void Test_ParseJavascriptWithHostLanguage()
        {
            // Get host-native Language via TreeSitterLanguagePackConverter
            var language = TreeSitterLanguagePackConverter.GetLanguage("javascript");
            Assert.NotNull(language);

            // Create upstream TreeSitter.DotNet parser
            using var parser = new TreeSitter.Parser();

            // Set the host-native language on the parser
            parser.Language = language;

            // Parse JavaScript source
            var source = "const x = 1;\n";
            using var tree = parser.Parse(source);

            var rootNode = tree.RootNode;
            Assert.NotNull(rootNode);

            // Verify the root node type is "program" (JavaScript AST root)
            Assert.Equal("program", rootNode.Type);
        }
    }
}

File: e2e/swift_e2e/Tests/TreeSitterLanguagePackE2ETests/CapsulePassthroughTests.swift
// Hand-written capsule-passthrough test (not fixture-generated); preserved across regen.

import XCTest
import Foundation
#if canImport(FoundationNetworking)
import FoundationNetworking
#endif
import TreeSitterLanguagePack
import SwiftTreeSitter

/// Hand-written e2e test for host-native Language passthrough.
/// Verifies that TreeSitterLanguagePack.getLanguage() returns a SwiftTreeSitter.Language
/// that is usable directly with the upstream SwiftTreeSitter parser.
final class CapsulePassthroughTests: XCTestCase {

  func testPassthroughPython() throws {
    // Parse Python code using capsule Language with upstream SwiftTreeSitter.Parser
    let pythonLang = try TreeSitterLanguagePack.getLanguage(name: "python")

    var parser = Parser()
    try parser.setLanguage(pythonLang)

    let source = "def greet(name):\n    return name\n"
    let tree = try parser.parse(source)

    let root = try XCTUnwrap(tree.rootNode)
    XCTAssertEqual(root.nodeType, "module")
  }

  func testPassthroughJavascript() throws {
    // Parse JavaScript code using capsule Language with upstream SwiftTreeSitter.Parser
    let jsLang = try TreeSitterLanguagePack.getLanguage(name: "javascript")

    var parser = Parser()
    try parser.setLanguage(jsLang)

    let source = "const x = 1;\n"
    let tree = try parser.parse(source)

    let root = try XCTUnwrap(tree.rootNode)
    XCTAssertEqual(root.nodeType, "program")
  }
}
